### PR TITLE
Fix some e2e tests issues + cleaning

### PR DESCRIPTION
**Pull request by ldevulder**
- Upload Cypress logs only on UI tests
- Use pure Go to share files through HTTP
- Move cypress tests files to `tests/`
- Some code cleanup

Verification runs - CLI:
- [OBS Dev - K3s - Elemental E2E tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3788068014)
- [OBS Stable - RKE2 - Elemental E2E tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3788070430)

Verification runs - UI:
- [OBS Dev - K3s - Elemental UI End-To-End tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3787764912)
- [OBS Dev - RKE2 - Elemental UI End-To-End tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3788059035)
- [OBS Stable - K3s - Elemental UI End-To-End tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3788062109)
- [OBS Stable - RKE2 - Elemental UI End-To-End tests with Rancher Manager](https://github.com/rancher/elemental/actions/runs/3788064744)

### CODE REVIEW

**Review by juadk — APPROVED** [review]
LGTM

**Review by mjura — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "elemental-docs",
```
Comment (juadk) [inline package.json:2]: I think this name can be changed too.

Comment (ldevulder) [reply package.json:2]: This is the package file for the docs part only, the one for Cypress has been moved in `tests` directory. So the `name` field is good.
This file can be removed in another PR when the docs part will be fully moved in a separate repository.

**Pull request merged**